R/proportional_hazards.R

```diff
@@ -206,6 +230,65 @@ predict_survival._coxnet <- function(object, new_data, ...) {
   predict_survival.model_fit(object, new_data = new_data, ...)
 }
 
+#' @export
+predict_raw._coxnet <- function(object, new_data, opts = list(), ...)  {
+  if (any(names(enquos(...)) == "newdata"))
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+  object$spec <- eval_args(object$spec)
+  opts$s <- object$spec$args$penalty
+  predict_raw.model_fit(object, new_data = new_data, opts = opts, ...)
+}
+
+#' @export
+multi_predict._coxnet <- function(object,
+                                  new_data,
+                                  type = NULL,
+                                  penalty = NULL,
+                                  ...) {
+    if (any(names(enquos(...)) == "newdata"))
+      rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+    dots <- list(...)
+
+    object$spec <- eval_args(object$spec)
+
+    if (is.null(penalty)) {
+      # See discussion in https://github.com/tidymodels/parsnip/issues/195
+      if (!is.null(object$spec$args$penalty)) {
+        penalty <- object$spec$args$penalty
+      } else {
+        penalty <- object$fit$lambda
+      }
+    }
+
+    if (type != "linear_pred"){
+      pred <- predict(object, new_data = new_data, type = type, ...,
+                      penalty = penalty, multi = TRUE)
+    } else {
+      pred <- predict(object, new_data = new_data, type = "raw",
+                      opts = dots, penalty = penalty, multi = TRUE)
+
+      # post-processing into nested tibble
+      param_key <- tibble(group = colnames(pred), penalty = penalty)
+      pred <- pred %>%
+        as_tibble() %>%
+        dplyr::mutate(.row = seq_len(nrow(pred))) %>%
+        tidyr::pivot_longer(
+          - .row,
+          names_to = "group",
+          values_to = ".pred_linear_pred"
+        )
+      pred <- dplyr::full_join(param_key, pred, by = "group") %>%
```

Comment:
I haven't run this, but it feels more like an `inner_join()`? Are you expecting rows from either side that don't match?

Reply:
good catch. I don't think so.